CMakeLists.txt

```diff
@@ -238,8 +243,25 @@ if(NOT ARB_USE_BUNDLED_RANDOM123)
     target_include_directories(ext-random123 INTERFACE ${RANDOM123_INCLUDE_DIR})
 endif()
 
-add_subdirectory(ext)
+# in the event we can find hwloc, just add it
+find_package(hwloc QUIET)
+add_library(ext-hwloc INTERFACE)
+if(hwloc_FOUND)
+    # We'd like to use the package syntax, here, yet if we do, we'd need to
+    # provide the find script to the system.
+    target_link_directories(ext-hwloc INTERFACE ${hwloc_LIBRARY_DIRS})
+    target_link_libraries(ext-hwloc INTERFACE ${hwloc_LIBRARY})
+    target_include_directories(ext-hwloc INTERFACE ${hwloc_INCLUDE_DIR})
+    target_compile_definitions(ext-hwloc INTERFACE ARB_HAVE_HWLOC)
+    target_link_libraries(arbor-private-deps INTERFACE ext-hwloc)
+else()
+    if(ARB_USE_HWLOC)
+        message(SEND_ERROR "Requested support for hwloc, but CMake couldn't find it.")
+    endif()
+endif()
```

Comment:
should we move this stuff into `ext/CMakeLists.txt`?

Reply:
Hmm, I am a bit torn here: `ext` is where we store 3rd party projects. This is a system dependency.